iohub/dataset_base.py

```diff
@@ -0,0 +1,81 @@
+from abc import ABC, abstractmethod
+from typing import Generator, Union
+
+import numpy as np
+from numpy.typing import ArrayLike
+
+
+class BaseFOV(ABC):
+    # NOTE: not using the `data` method from ngff Position
```

Comment:
Then what is the intended way to access voxel values?

Reply:
p.s. the `data` property is not exactly 'intended' either...

Reply:
I thought of accessing through `__getitem__` and having additional methods to access the multi-scale in ome-zarr datasets. We would have to have a policy to select an array when multiple keys are present.

Reply:
In practice most of the acquisition data formats (TIFF, clear control) do not have multi-scale. Should the key always be '0' for them?

Reply:
Oh wait did you mean that `__getitem__` directly slices the lowest resolution array? So getting the whole FOV would be `BaseFOV(...)[:]`?

Reply:
I would slice the **highest** resolution.

It would behave sort of like a zarr.
I'm not completely sure if it should load into memory or allow returning any kind of array (e.g. dask array).

I'm more in favor of the latter, and the user should call ```np.asarray``` when it must be loaded into memory, which is not always the case because `numpy` should process any `arraylike` data.

Reply:
Got it, so internally it would be like this (for NGFF)?

```py
class Position(...):
    def __getitem__(self, key):
        return self["0"][key]
```

Reply:
Yes, we could even allow any key, and use "0" or the first in lexicographic order when more than one key is present.

Reply:
I don't think we currently use any other format with pyramids or other data structure that needs multiple array keys in the same FOV. However it does call for more change in the NGFF API since it also does writing, since if `__getitem__` operates on one of the arrays in the FOV, `__setitem__` should match that behavior.

Reply:
Yes, it will need to be refactored.